src/api_server/src/resource/mod.rs

```diff
@@ -106,8 +105,13 @@ pub(crate) async fn get_secret_resource(
         .encrypt(nonce, resource_byte.as_slice())
         .map_err(|e| anyhow!("AES encrypt Resource payload failed: {:?}", e))?;
 
-    let rsa_pub_key = RsaPublicKey::from_public_key_pem(&tee_pub_key.k)
-        .map_err(|e| anyhow!("Parse TEE pubkey PEM failed: {:?}", e))?;
+    let n = BigUint::from_str_radix(&tee_pub_key.k_mod, 10)
+        .map_err(|e| anyhow!("Parse TEE pubkey modulus failed: {:?}", e))?;
+    let e = BigUint::from_str_radix(&tee_pub_key.k_exp, 10)
+        .map_err(|e| anyhow!("Parse TEE pubkey exponent failed: {:?}", e))?;
```

Comment:
Both k_mod and k_exp are `base64` formatted so I think you want to decode that first.

Also, I'm trying to push https://github.com/virtee/kbs-types/pull/19 to make it use JWA compliant JSON payloads, if we decide to drop the more generic `k` field to describe a key.

Reply:
Ah that makes sense. It wasn't documented on the types what the encoding is supposed to be, afaict. Will update this on and the AA PR accordingly. 

the n/e renaming should be transparent to this code, right?

Reply:
> Ah that makes sense. It wasn't documented on the types what the encoding is supposed to be, afaict. Will update this on and the AA PR accordingly.
> 

Talking about documentation, would you mind updating the kbs_protocol as well (when/if the pending PR gets merged) at https://github.com/confidential-containers/kbs/blob/main/docs/kbs_attestation_protocol.md#public-key ?

> the n/e renaming should be transparent to this code, right?

It should. I did not change the structure type, only the JSON serialization format.

Reply:
> > Talking about documentation, would you mind updating the kbs_protocol as well (when/if the pending PR gets merged) at https://github.com/confidential-containers/kbs/blob/main/docs/kbs_attestation_protocol.md#public-key ?
> 
> yup, will do. We'd need a kbs-types release for that too, right?

Thanks for the doc fix.
We'd need a new `kbs-types` release, provided the PR is accepted. The new release will include the `AzSnpVtpm` type as well, so would you prefer to wait for that release or to just go ahead with this set of PRs?

Reply:
> > Thanks for the doc fix. We'd need a new `kbs-types` release, provided the PR is accepted. The new release will include the `AzSnpVtpm` type as well, so would you prefer to wait for that release or to just go ahead with this set of PRs?
> 
> Not sure about the kbs-types release cadence. I'd recommend to go ahead with this set of fixes, limiting the scope to an update to kbs-types#v0.2.0 and then deal with `n/e` and `AzSnpVtpm` individually. There is already a lot of untangling and cognitive gymnastics required to update the kbs-types dep in concert across three repos.

Fine with me. Moving to 0.3.0 should be less painful as hopefully there won't be any API breaking changes.